src/search_internal.h

```diff
@@ -106,9 +107,10 @@ struct search_iterator::InternalData {
 
     Xapian::Document get_document() {
         if ( !document_fetched ) {
-            if (iterator != mp_mset->end()) {
-                _document = iterator.get_document();
+            if (iterator == mp_mset->end()) {
+                throw EntryNotFound("Cannot get entry for end iterator");
```

Comment:
Is `EntryNotFound` type of error appropriate in this case? Dereferencing an end iterator can be a result of out-of-bounds access, which is a bug rather than an outcome of an unsatisfiable query.

Reply:
Dereferencing a end iterator is undefined.
Now we throw a std::runtime_error.